services/grpcService.ts

```diff
@@ -203,27 +212,41 @@ export class GrpcBlockchainClient implements BlockchainClient {
     return details
   };
 
-  public async subscribeTransactions (
-    addresses: string[],
-    onTransactionNotification: (txn: GrpcTransaction.AsObject) => any,
-    onMempoolTransactionNotification: (txn: GrpcTransaction.AsObject) => any,
-    networkSlug: string
-  ): Promise<void> {
-    const createTxnStream = async (): Promise<void> => {
-      const client = this.getClientForNetworkSlug(networkSlug)
+  public async subscribeAddressesAddTransactions (addresses: Address[]): Promise<void> {
+    if (addresses.length === 0) return
+
+    const addressesAlreadySubscribed = addresses.filter(address => Object.keys(this.subscribedAddresses).includes(address.address))
+    if (addressesAlreadySubscribed.length === addresses.length) return
+    addressesAlreadySubscribed.forEach(address => {
+      console.log(`This address was already subscribed: ${address.address}`)
+    })
+    addresses = addresses.filter(address => !addressesAlreadySubscribed.includes(address))
+
+    const addressesInClients: KeyValueT<Address[]> = {}
+    this.availableNetworks.forEach(networkSlug => {
+      addressesInClients[networkSlug] = []
+    })
+    addresses.forEach(address => {
```

Comment:
Actually I think this block from 225 to 233 could be in a separate auxiliary method (maybe in `utils/index` or `services/addressService`) like `groupAddressesByNetwork`

Reply:
indeed, much better, done